Parse/Vehicle.swift

```diff
@@ -65,6 +67,34 @@ struct Vehicle {
         }
     }
     
+    /**
+     Estimates the current position of this Vehicle based on the previous Updates received.
+     - Returns: A Point corresponding to the current estimated position for this Vehicle.
+     */
+    func estimateCurrentPosition() -> Point {
+        let feet_per_second = (last_update.speed / 60) * 5280 // TODO: Assumes speed is in mph; could be km/h
+        let distance = feet_per_second * secondsSince(update: last_update) // In feet
+        let start = Point(update: last_update)
+        var startIndex = 0
+        for i in 0..<last_update.route.points.count {
+            if last_update.route.points[i] == start {
+                startIndex = i
+                break
+            }
+        }
+        var elapsedDistance = 0.0
+        var endPoint = start
+        for i in startIndex..<last_update.route.points.count {
+            elapsedDistance += start.distanceFrom(p: last_update.route.points[i])
+            if elapsedDistance > distance {
+                // TODO: Determine intermediate points based on excess distance
+                endPoint = last_update.route.points[i]
+                break
+            }
+        }
+        return endPoint
+    }
+    
```

Comment:
Why iterate through last_update? shouldn't it be the lowest index or highest index?

Reply:
We iterate through last_update.route.points in order to find a point which point in the route is closest to the shuttle's current position. Although, the shuttle's position may not align with a point in the route exactly, so instead of checking for equality we should keep track of the current shortest distance.

Reply:
Fine for now, probably a better way to do this.